src/Datasets.jl

```diff
@@ -27,45 +37,104 @@ function set_band_names(filename::String, band_names::Vector)
+        outDataset = ArchGDAL.create(
+            output_files[_o], driver=ArchGDAL.getdriver("ENVI"), width=x_len,
+            height=y_len, nbands=output_bands[_o], dtype=Float64
+        )
         ArchGDAL.setproj!(outDataset, ArchGDAL.getproj(reference_dataset))
-        @info "Output Image Size (x,y,b): $x_len, $y_len, $output_bands.  Creating output fractional cover dataset."
+        @info "Output Image Size (x,y,b): $x_len, $y_len, $output_bands.
+        Creating output fractional cover dataset."
         try
-            ArchGDAL.setgeotransform!(outDataset, ArchGDAL.getgeotransform(reference_dataset))
+            ArchGDAL.setgeotransform!(
+                outDataset, ArchGDAL.getgeotransform(reference_dataset)
+            )
         catch e
-            println("No geotransorm avaialble, proceeding without")
+            println("No geotransorm available, proceeding without")
         end
 
         for _b in 1:length(output_bands)
-            ArchGDAL.setnodatavalue!(ArchGDAL.getband(outDataset,_b), -9999)
+            ArchGDAL.setnodatavalue!(ArchGDAL.getband(outDataset, _b), -9999)
         end
 
         output = zeros(x_len, y_len, output_bands[_o]) .- 9999
-        ArchGDAL.write!(outDataset, output, [1:size(output)[end];], 0, 0, size(output)[1], size(output)[2])
+        ArchGDAL.write!(
+            outDataset, output, [1:size(output)[end];], 0, 0, size(output)[1],
+            size(output)[2]
+        )
 
         outDataset = nothing
-
     end
 end
 
+"""
+    write_results(output_files::Vector{String}, output_bands::Vector, x_len::Int64,
+                  y_len::Int64, results, args)
+
+Write processed results to specified output ENVI raster files, including primary outputs,
+uncertainty estimates, and complete fractions, based on the provided parameters.
+
+# Arguments
+- `output_files::Vector{String}`: File paths of the output datasets to be created or
+updated. Each entry corresponds to a separate output raster.
+- `output_bands::Vector`: Number of bands for each output dataset. The length of this
+vector must match the number of output files.
+- `x_len::Int64`: Width of the output datasets.
+- `y_len::Int64`: Height of the output datasets.
+- `results`: A collection of results, where each result is expected to be a tuple or
+similar structure. Specific elements contain the data to be written to the output datasets.
+- `args`: An object or structure containing additional parameters that dictate the
+writing behavior. Available options are:
+    - `args.n_mc`: If greater than 1, an uncertainty output is also written to the second
+    output file.
+    - `args.write_complete_fractions`: If set to 1, write complete fractions to the
+    subsequent output file.
+
+# Notes
+- The primary output is initialized with a no-data value of `-9999` and written to the
+  first output file specified in `output_files`.
+- Each output array is permuted to match the desired dimensions before writing.
+"""
```

Comment:
This function seems like it has some pretty specific usage but it's never used by `SpectralUnmixing.jl` (unlike `write_line_results()` or `load_line()`) or used by `unmix.jl` even though it's exported.

Reply:
It's a choice - this could be used cleanly, if you wanted to write full (rather than line-by-line results).  Both are valid, so let's keep this...it may be valuable to some users.